File: frontend/src/types/platform.ts
export type CategorySummary = {
  key: "P1" | "P2" | "P3" | "P4";
  name: string;
  capability_count: number;
  status: string;
  capabilities: CapabilitySummary[];
};

export type CapabilitySummary = {
  key: string;
  name: string;
  brand_key: string;
  category_key: "P1" | "P2" | "P3" | "P4";
  status: string;
};

export type BrandSummary = {
  key: string;
  name: string;
  tagline: string;
  active_category_count: number;
  categories: CategorySummary[];
};

export type ProjectSummary = {
  key: string;
  name: string;
  brand_key: string;
  category_key: "P1" | "P2" | "P3" | "P4";
  status: string;
  integration_key?: string | null;
  entry_path?: string | null;
};

export type IntegrationHealth = {
  service: string;
  status: "AVAILABLE" | "UNAVAILABLE";
  entry_path: string;
  message: string;
};

export type FeedbackSummary = {
  project: string;
  mapped_project_key: string | null;
  mapped_project_name: string | null;
  brand_key: string | null;
  category_key: "P1" | "P2" | "P3" | "P4" | null;
  status: string;
  original_manual_time: string;
  current_processing_time: string;
  business_feedback: string;
  iteration_need: string;
  updated_by: string;
  updated_at: string;
};

export type DashboardSummary = {
  brands: BrandSummary[];
  projects: ProjectSummary[];
  kpis: Record<string, number | string>;
};

export type ScheduleItem = {
  key: string;
  name: string;
  brand_key: string;
  status: string;
  milestone: string;
  risk: string;
};

export type EmptyModule = {
  route: string;
  title: string;
  status: string;
  message: string;
};

File: backend/schemas/platform.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class CategorySummary(BaseModel):
    model_config = ConfigDict(frozen=True)

    key: str = Field(pattern=r"^P[1-4]$")
    name: str
    capability_count: int = Field(ge=0)
    status: str
    capabilities: tuple["CapabilitySummary", ...] = ()


class CapabilitySummary(BaseModel):
    model_config = ConfigDict(frozen=True)

    key: str
    name: str
    brand_key: str
    category_key: str = Field(pattern=r"^P[1-4]$")
    status: str


class BrandSummary(BaseModel):
    model_config = ConfigDict(frozen=True)

    key: str
    name: str
    tagline: str
    active_category_count: int = Field(ge=0)
    categories: tuple[CategorySummary, ...]


class ProjectSummary(BaseModel):
    model_config = ConfigDict(frozen=True)

    key: str
    name: str
    brand_key: str
    category_key: str = Field(pattern=r"^P[1-4]$")
    status: str


class FeedbackSummary(BaseModel):
    model_config = ConfigDict(frozen=True)

    project: str
    mapped_project_key: str | None = None
    mapped_project_name: str | None = None
    brand_key: str | None = None
    category_key: str | None = Field(default=None, pattern=r"^P[1-4]$")
    status: str
    original_manual_time: str
    current_processing_time: str
    business_feedback: str
    iteration_need: str
    updated_by: str
    updated_at: str


class DashboardSummary(BaseModel):
    model_config = ConfigDict(frozen=True)

    brands: tuple[BrandSummary, ...]
    projects: tuple[ProjectSummary, ...]
    kpis: dict[str, int | str]


class ScheduleItem(BaseModel):
    model_config = ConfigDict(frozen=True)

    key: str
    name: str
    brand_key: str
    status: str
    milestone: str
    risk: str


class EmptyModule(BaseModel):
    model_config = ConfigDict(frozen=True)

    route: str
    title: str
    status: str
    message: str
